File: packages/cli/test/fixtures/ranking-corpora/go/internal/handlers/create.go
// Package handlers implements the HTTP-ish handler layer for the fixture
// corpus's tiny in-memory "user service" example.
package handlers

import (
	"errors"

	"github.com/lien-fixtures/ranking-go/internal/util"
)

// CreateUser validates and records a new user by name, logging the outcome
// through the shared util package.
func CreateUser(name string) error {
	if name == "" {
		err := errors.New("name must not be empty")
		util.LogError("CreateUser", err)
		return err
	}
	util.LogInfo("created user " + name)
	return nil
}

File: packages/cli/test/fixtures/ranking-corpora/typescript/src/handlers/create.ts
import { logError, logInfo } from '../util/logger.js';

/** Validates and records a new user by name. */
export function createUser(name: string): void {
  if (!name) {
    logError('createUser', new Error('name must not be empty'));
    return;
  }
  logInfo(`created user ${name}`);
}
